pilot/pkg/model/service.go

```diff
@@ -655,6 +662,20 @@ func (s *Service) GetServiceAddressForProxy(node *Proxy) string {
 	return s.Address
 }
 
+// GetTLSModeFromEndpointLabels checks if the labels contain one of the TLS mode labels
+// and if so, returns the appropriate constant
+func GetTLSModeFromEndpointLabels(labels map[string]string) string {
```

Comment:
nit. May be returning an enum instead.

Reply:
See the explanation below for why we are returning the string. This is to let others extend the modes and use EnvoyFilter for customization. If you change this to an enum, then it wont work.